### ISSUE TITLE
BUG: Unhandled error in plugin discover: 'ascii' codec can't encode character u'\u2013'

### ISSUE THREAD

**Issue by programatix** [Flexget/Flexget#1113]
Hi,

Just updated flexget to version 2.0.8 and I'm getting the following error,
```
2016-04-30 02:20 DEBUG    est_series_tvmaze NyaaTorrents    show_name: Soukyuu no Fafner Dead Aggressor – Exodus
2016-04-30 02:20 DEBUG    est_series_tvmaze NyaaTorrents    series_name: Soukyuu no Fafner Dead Aggressor – Exodus
2016-04-30 02:20 DEBUG    api_tvmaze    NyaaTorrents    returning search params for series lookup: {u'tvdb_id': None, u'tvmaze_id': None, u'name': u'Soukyuu no Fafner Dead Aggressor \u2013 Exodus', u'tvrage_id': None}
2016-04-30 02:20 DEBUG    api_tvmaze    NyaaTorrents    searching db tvmaze_series for the values [(u'tvdb_id', None), (u'tvmaze_id', None), (u'name', u'Soukyuu no Fafner Dead Aggressor \u2013 Exodus'), (u'tvrage_id', None)]
2016-04-30 02:20 DEBUG    api_tvmaze    NyaaTorrents    did not find exact match for series Soukyuu no Fafner Dead Aggressor – Exodus in cache, looking in search table
2016-04-30 02:20 DEBUG    api_tvmaze    NyaaTorrents    searching lookup table using title Soukyuu no Fafner Dead Aggressor – Exodus
2016-04-30 02:20 CRITICAL task          NyaaTorrents    BUG: Unhandled error in plugin discover: 'ascii' codec can't encode character u'\u2013' in position 33: ordinal not in range(128)
Traceback (most recent call last):
  File "/opt/lib/python2.7/site-packages/flexget/task.py", line 442, in __run_plugin
    return method(*args, **kwargs)
  File "/opt/lib/python2.7/site-packages/flexget/event.py", line 23, in __call__
    return self.func(*args, **kwargs)
  File "/opt/lib/python2.7/site-packages/flexget/plugins/input/discover.py", line 276, in on_task_input
    entries = self.estimated(entries, estimation_mode)
  File "/opt/lib/python2.7/site-packages/flexget/plugins/input/discover.py", line 192, in estimated
    est_date = estimator.estimate(entry)
  File "/opt/lib/python2.7/site-packages/flexget/plugins/estimators/est_released.py", line 29, in estimate
    estimate = estimator(entry)
  File "/opt/lib/python2.7/site-packages/flexget/plugins/estimators/est_release_series_tvmaze.py", line 53, in estimate
    episode = lookup(**kwargs)
  File "/opt/lib/python2.7/site-packages/flexget/utils/database.py", line 34, in wrapper
    return func(*args, **kwargs)
  File "/opt/lib/python2.7/site-packages/flexget/plugins/api_tvmaze.py", line 527, in episode_lookup
    series = APITVMaze.series_lookup(session=session, only_cached=only_cached, **lookup_params)
  File "/opt/lib/python2.7/site-packages/flexget/utils/database.py", line 31, in wrapper
    return func(*args, **kwargs)
  File "/opt/lib/python2.7/site-packages/flexget/plugins/api_tvmaze.py", line 471, in series_lookup
    prepared_params = prepare_lookup_for_pytvmaze(**lookup_params)
  File "/opt/lib/python2.7/site-packages/flexget/plugins/api_tvmaze.py", line 429, in prepare_lookup_for_pytvmaze
    prepared_params['show_name'] = native_str(title) if title else None
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2013' in position 33: ordinal not in range(128)
```

Any help would be greatly appreciated. Thanks.

**Comment by programatix** [comment]
It seems that the `series lookup` converted the `-` into `\u2013` as shown by the log above,
```
2016-04-30 02:20 DEBUG    est_series_tvmaze NyaaTorrents    series_name: Soukyuu no Fafner Dead Aggressor – Exodus
2016-04-30 02:20 DEBUG    api_tvmaze    NyaaTorrents    returning search params for series lookup: {u'tvdb_id': None, u'tvmaze_id': None, u'name': u'Soukyuu no Fafner Dead Aggressor \u2013 Exodus', u'tvrage_id': None}
```

**Comment by crawln45** [comment]
that would be because the input had it in the title. @liiight your baby is becoming more hateful

**Comment by programatix** [comment]
I also have other title which has `-` in it, example,
```
"Ghost in the Shell Arise - Alternative Architecture"
"Sidonia no Kishi - Daikyuu Wakusei Seneki"
"Yamada-kun and the Seven Witches"
```

and they are ok. So, most likely the `\u2013` is from tvmaze db.

**Comment by crawln45** [comment]
yes but seems that that title has a unicode```\u2013``` in it. not a '-' but a [–](http://www.fileformat.info/info/unicode/char/2013/index.htm)

**Comment by programatix** [comment]
That's the very first thought that came to my mind. I have deleted it and retyped it in notepad++ and the same error is encountered in the next run.

**Comment by crawln45** [comment]
Interesting. Seems something isn't converting unicode properly.

**Comment by programatix** [comment]
Ok, I think flexget is not using the latest config.yml I pointed it to use. In the DEBUG, it did state that it detected new config.yml but when discovering, it still uses old list. What I did was to comment out the problematic title and the error still exists.

**Comment by crawln45** [comment]
what's your config look like see if i can replicate it.

**Comment by crawln45** [comment]
if it's the daemon. then you have to reload the config after a change flexget daemon reload

**Comment by programatix** [comment]
I'm pretty sure I've stopped the daemon. Let me retest again.

**Comment by programatix** [comment]
Yes, flexget is using old list even though I've updated the config.yml. I can see removed title still being searched. No matter how I execute flexget, I get
```
2016-04-30 03:53 VERBOSE  task_queue                    There are 1 tasks to execute. Shutdown will commence when they have completed.
```

**Comment by crawln45** [comment]
what's the config look like.

**Comment by programatix** [comment]
Something like this (extracted). Btw, you'll see the weird `begin` on some title because `discover` won't look into them if they are new title.
```
tasks:
  NyaaTorrents:
    discover:
      what:
        - emit_series:
            from_start: yes
            backfill: yes
      from:
        - nyaa: anime
    template: [cartoon]
    
templates:
  cartoon:
    series:
      settings:
        cartoon:
          propers: 3 days
          quality: <=720p
          set:
            path: '/mnt/storage1/Videos/Cartoon/{{series_name}}'
        cartoon_seasoned:
          propers: 3 days
          quality: <=720p
          set:
            path: '/mnt/storage1/Videos/Cartoon/{{series_name}}'
        cartoon_anime:
          propers: 3 days
          quality: <=720p
          identified_by: sequence
          sequence_regexp:
            - (\d+)
            - S(\d+) (\d+)
          from_group: horriblesubs
          set:
            path: '/mnt/storage1/Videos/Cartoon/{{series_name}}'
        anime_various:
          propers: 3 days
          quality: <=720p
          identified_by: sequence
          sequence_regexp:
            - (\d+)
            - S(\d+) (\d+)
          from_group: horriblesubs
          set:
            path: '/mnt/storage1/Videos/TV Series/Various/{{series_name}}'
            
      cartoon:
        - Pokemon The Series XY:
            from_group: c-w
            path: '/mnt/storage1/Videos/Cartoon/Pokemon/17 {{series_name}}'
            
      cartoon_anime:
        - Hunter X Hunter
        - Fairy Tail S2
        - Boku dake ga Inai Machi
        
      anime_various:
        - Terra Formars S2:
            begin: 01
        - Ajin:
            begin: 01
            from_group: DeadFish
        - "JoJo's Bizarre Adventure"
        # - "Soukyuu no Fafner Dead Aggressor - Exodus"

    template: transmissionRPC
    transmission:
      addpaused: no
      ratio: 0
```

**Comment by programatix** [comment]
Ok, for some weird reason flexget is now using the correct list now. Even the `"Soukyuu no Fafner Dead Aggressor - Exodus"` is now searched correctly with no error.

So sorry for the trouble caused.

**Comment by crawln45** [comment]
Fixed e5ab31c
